airbyte-featureflag/src/main/kotlin/Context.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.featureflag
+
+/**
+ * Context abstraction around LaunchDarkly v6 context idea
+ *
+ * I'm still playing around with this.  Basically the idea is to define our own custom context types
+ * (by implementing this sealed interface) to ensure that we are consistently using the same identifiers
+ * throughout the code.
+ *
+ * @property [kind] determines the kind of context the implementation is,
+ * must be consistent for each type and should not be set by the caller of a context
+ * @property [key] is the unique identifier for the specific context, e.g. a user-id or workspace-id
+ */
+sealed interface Context {
+    val kind: String
```

Comment:
`kind` feels a bit redundant with the actual `data class`. The overhead isn't big, I was wondering if you considered not having it?

Reply:
Originally I didn't have it, for this very reason, it's redundant. 

The problem I ran into is that LaunchDarkly wants a string passed to their [ContextKind.of](https://javadoc.io/doc/com.launchdarkly/launchdarkly-java-server-sdk/latest/com/launchdarkly/sdk/ContextKind.html) method (a requirement for their v6 sdk) which ended up with the code having to check which type of `Context` was provided and then assigning that value to a string to pass to `ContextKind.of`
```kotlin
private fun Context.toLDContext(): LDContext {
  val kind = when(this) {
    is Workspace -> "workspace"
    is User -> "user"
  }
  val builder = LDContext.builder(ContextKind.of(kind), key)
  // ...
}
```

I found it cleaner to place this unique-string-identifer into the `data class` itself.